### PR TITLE
Feat: mobile auth

### PR DESCRIPTION
**Pull request by mariusjagminas**

### CODE REVIEW

File: src/phone-auth/phone-auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { Injectable } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Strategy, VerifiedCallback } from 'passport-custom';
+import { Repository } from 'typeorm';
+import dayjs from 'dayjs';
+
+import { User } from '../users/user.entity';
+
+const SMS_CODE_EXPIRATION = 60 * 1000;
```
Comment (kowalski) [inline src/phone-auth/phone-auth.guard.ts:11]: 1 minutes is a short time :) give it 10 maybe?

File: src/users/user.entity.ts [hunk 2]
```diff
@@ -11,15 +11,24 @@ export class User {
...
   status: string;
 
+  @Column({ unique: true, nullable: true })
+  phone: string;
+
+  @Column({ default: '' })
+  code: string;
+
+  @Column({ default: '' })
+  codeGeneratedAt: string;
```
Comment (kowalski) [inline src/users/user.entity.ts:30]: this field represents a date so it should be stored as date in database (not string)

```
  @CreateDateColumn({ nullable: true })
  public codeGeneratedAt: Date;
```

```
import {
  CreateDateColumn,
  ...
} from 'typeorm';

File: src/phone-auth/phone-auth.controller.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import { SignUpResponseDto } from '../users/user.interface';
+import { PhoneLogInDto, PhoneSignUpDto } from './phone-auth.interface';
+import { PhoneAuthService } from './phone-auth.service';
+import { UserSessionDto } from '../auth/auth.interface';
+
+@Controller('phone-auth')
+export class PhoneAuthController {
+  constructor(private phoneAuthService: PhoneAuthService) {}
+  @Post('sign-up')
```
Comment (kowalski) [inline src/phone-auth/phone-auth.controller.ts:12]: I would prefer this endpoint to be called something like "send-code"

File: src/phone-auth/phone-auth.interface.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { IsMobilePhone, IsNotEmpty } from 'class-validator';
+
+export class PhoneSignUpDto {
+  @IsMobilePhone()
```
Comment (kowalski) [inline src/phone-auth/phone-auth.interface.ts:4]: I think the right validator here is:
```
@IsPhoneNumber('ZZ')
```

source: https://github.com/typestack/class-validator

Comment (kowalski) [reply src/phone-auth/phone-auth.interface.ts:4]: also I I'm not sure if you don't need to combine it with "@IsNotEmpty()".

**Review by kowalski — CHANGES_REQUESTED** [review]

File: src/phone-auth/phone-auth.guard.ts (hunk 1 above)
Comment (mariusjagminas) [reply src/phone-auth/phone-auth.guard.ts:11]: ok iI will

File: src/users/user.entity.ts (hunk 2 above)
Comment (mariusjagminas) [reply src/users/user.entity.ts:30]: Yep. That should be date

File: src/phone-auth/phone-auth.controller.ts (hunk 3 above)
Comment (mariusjagminas) [reply src/phone-auth/phone-auth.controller.ts:12]: ok. What about 'phone-auth' , maybe 'mobile-auth' would be a better name?

File: src/phone-auth/phone-auth.interface.ts (hunk 4 above)
Comment (mariusjagminas) [reply src/phone-auth/phone-auth.interface.ts:4]: I will check that

File: src/phone-auth/phone-auth.controller.ts (hunk 3 above)
Comment (kowalski) [reply src/phone-auth/phone-auth.controller.ts:12]: hmm, this endpoints sends a code to a given number
it's all it does 
I would just call it 'send-code', really

Comment (kowalski) [reply src/phone-auth/phone-auth.controller.ts:12]: phone-auth is already the route of the controller, so the whole path becomes
/phone-auth/send-code

File: src/phone-auth/phone-auth.interface.ts (hunk 4 above)
Comment (mariusjagminas) [reply src/phone-auth/phone-auth.interface.ts:4]: @IsMobilePhone () is a valid validator, it also checks that it is not empty.

File: src/phone-auth/phone-auth.controller.ts (hunk 3 above)
Comment (mariusjagminas) [reply src/phone-auth/phone-auth.controller.ts:12]: so probably `/phone-auth/log-in` route  should be renamed as  `verify-code`

File: src/phone-auth/phone-auth.interface.ts (hunk 4 above)
Comment (kowalski) [reply src/phone-auth/phone-auth.interface.ts:4]: but does @IsMobilePhone() enforce that the country code is present ? So like +48 XXX XXX XXX (X)?

File: src/phone-auth/phone-auth.controller.ts (hunk 3 above)
Comment (kowalski) [reply src/phone-auth/phone-auth.controller.ts:12]: yes, I like it

**Review by kowalski — APPROVED** [review]
Note for next work:

We are now removing phoneNumber from user profile. We probably should add it back in GET request, but not in PATCH request. I think the in GET we should return email and phoneNumber of the user. Just not support changing it in PATCH.

**Pull request merged**